TailLight/device.cpp

```diff
@@ -19,6 +19,32 @@ VOID EvtSetBlackTimer(_In_ WDFTIMER  Timer) {
     KdPrint(("TailLight: EvtSetBlackTimer end\n"));
 }
 
+NTSTATUS EvtSelfManagedIoInit(WDFDEVICE device) {
+    // Initialize tail-light to black to have control over HW state
+    WDF_TIMER_CONFIG timerCfg = {};
+    WDF_TIMER_CONFIG_INIT(&timerCfg, EvtSetBlackTimer);
+
+    WDF_OBJECT_ATTRIBUTES attribs = {};
+    WDF_OBJECT_ATTRIBUTES_INIT(&attribs);
+    attribs.ParentObject = device;
```

Comment:
This keeps the object alive for as long as the device is alive. Can we shorten the lifetime of attribs down?

Reply:
Not really. The driver also creates a queue for IOCTL processing, but it has the same lifetime as the device.

I though about deleting the timer object in the `EvtSetBlackTimer` callback, but that will according to the documentation lead to a [deadlock](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wdftimer/nc-wdftimer-evt_wdf_timer#remarks). This can be worked around by scheduling a work item to delete the timer, but that sounds like a lot of complexity for little gain.

Reply:
Object cleanup would probably have been easier if we'd been able to replace the timer with a work item. However, my attempts on doing so have so far been unsuccessful.